src/h5web/vis-packs/core/line/MappedLineVis.tsx

```diff
@@ -53,27 +47,25 @@ function MappedLineVis(props: Props): ReactElement {
   } = useLineConfig((state) => state, shallow);
 
   const value = useDatasetValue(valueDataset);
-  const baseDataArray = useBaseArray(value, dims);
-  const dataArray = useMappedArray(baseDataArray, dimMapping);
+  const dataArray = useArrays(value, dims, dimMapping);
 
   const errors = useDatasetValue(errorsDataset);
-  const baseErrorsArray = useBaseArray(errors, dims);
-  const errorArray = useMappedArray(baseErrorsArray, dimMapping);
+  const errorArray = useArrays(errors, dims, dimMapping);
 
+  const auxArrays = useAuxDatasetArrays(auxDatasets, dims, dimMapping);
+
+  // Disable `autoScale` for 1D datasets (baseArray and dataArray are the same)
   useEffect(() => {
     // Disable `autoScale` for 1D datasets without errors
-    disableAutoScale(baseDataArray.shape.length <= 1 && !errors);
-  }, [baseDataArray.shape, disableAutoScale, errors]);
-
-  const dataValues = (autoScale ? dataArray : baseDataArray).data as number[];
-  const errorValues = autoScale
-    ? errorArray && (errorArray.data as number[])
-    : baseErrorsArray && (baseErrorsArray.data as number[]);
+    disableAutoScale(dataArray.base.shape.length <= 1 && !errors);
+  }, [dataArray.base.shape, disableAutoScale, errors]);
 
-  const dataDomain = useDomain(
-    dataValues,
-    yScaleType,
-    showErrors || !autoScale ? errorValues : undefined
+  const dataDomain = useAutoScaledDomain(
+    dataArray,
+    showErrors ? errorArray : undefined,
+    auxArrays,
+    autoScale,
+    yScaleType
```

Comment:
Rather than making a new hook only for the `auxArrays`, I crossed the bridge and put the `autoScale` code in it as well.

Again, I welcome suggestions for the name.